### PR TITLE
python311Packages.python-ldap: 3.4.3 -> 3.4.4

### PR DESCRIPTION
**Pull request by mweinelt**
https://github.com/python-ldap/python-ldap/releases/tag/python-ldap-3.4.4

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 268179` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>powerdns-admin</li>
  </ul>
</details>
<details>
  <summary>25 packages built:</summary>
  <ul>
    <li>adenum</li>
    <li>autofs5</li>
    <li>etebase-server (python311Packages.etebase-server)</li>
    <li>freeipa</li>
    <li>ldapmonitor</li>
    <li>python310Packages.django-auth-ldap</li>
    <li>python310Packages.django-auth-ldap.dist</li>
    <li>python310Packages.dogtag-pki</li>
    <li>python310Packages.dogtag-pki.dist</li>
    <li>python310Packages.etebase-server</li>
    <li>python310Packages.ldappool</li>
    <li>python310Packages.ldappool.dist</li>
    <li>python310Packages.python-ldap</li>
    <li>python310Packages.python-ldap.dist</li>
    <li>python311Packages.django-auth-ldap</li>
    <li>python311Packages.django-auth-ldap.dist</li>
    <li>python311Packages.dogtag-pki</li>
    <li>python311Packages.dogtag-pki.dist</li>
    <li>python311Packages.ldappool</li>
    <li>python311Packages.ldappool.dist</li>
    <li>python311Packages.python-ldap</li>
    <li>python311Packages.python-ldap.dist</li>
    <li>silenthound</li>
    <li>sssd</li>
    <li>tandoor-recipes</li>
  </ul>
</details>

File: pkgs/development/python-modules/python-ldap/default.nix [hunk 1]
```diff
@@ -2,27 +2,41 @@
...
-, pyasn1
-, pyasn1-modules
-, pythonAtLeast
 , pythonOlder
-, pytestCheckHook
+
+# build-system
+, setuptools
+
+# native dependencies
```
Comment (atalii) [inline pkgs/development/python-modules/python-ldap/default.nix:10]: I'd get rid of these comments. All this information is present in the arguments to buildPythonPackage, and if these comments ever get out of date they may do more harm than good.

File: pkgs/development/python-modules/python-ldap/default.nix [hunk 2]
```diff
@@ -2,27 +2,41 @@
...
 , openldap
 , cyrus_sasl
+
+# dependencies
+, pyasn1
+, pyasn1-modules
+
+# tests
+, pytestCheckHook
```
Comment (atalii) [inline pkgs/development/python-modules/python-ldap/default.nix:19]: Probably best to remove whitespace and comments here for the same reason.

File: pkgs/development/python-modules/python-ldap/default.nix [hunk 3]
```diff
@@ -53,7 +67,9 @@ buildPythonPackage rec {
   doCheck = !stdenv.isDarwin;
 
   meta = with lib; {
+    changelog = "https://github.com/python-ldap/python-ldap/releases/tag/python-ldap-${version}";
     description = "Python modules for implementing LDAP clients";
+    downloadPage = "https://github.com/python-ldap/python-ldap";
     homepage = "https://www.python-ldap.org/";
     license = licenses.psfl;
   };
```
Comment (atalii) [inline pkgs/development/python-modules/python-ldap/default.nix:75]: Is anyone in particular maintaining this? Would you be okay volunteering?

**Review by atalii — CHANGES_REQUESTED** [review]
Mostly LGTM! I'm still running nixpkgs-review everything that depends on this, but I expect that should pass. Should this be backported to 23.05?

File: pkgs/development/python-modules/python-ldap/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/python-ldap/default.nix:10]: Same comment style as in the python documentation for nixpkgs, coincidentally contributed by me. https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-function-buildpythonpackage-function

To me organizing them like this is preferable to making it an opaque list without context.

File: pkgs/development/python-modules/python-ldap/default.nix (hunk 3 above)
Comment (mweinelt) [reply pkgs/development/python-modules/python-ldap/default.nix:75]: I'm basically the fallback maintainer for all python packages anyway, so it wouldn't change anything.

File: pkgs/development/python-modules/python-ldap/default.nix (hunk 1 above)
Comment (atalii) [reply pkgs/development/python-modules/python-ldap/default.nix:10]: Oh, look at that. Sorry! Will resolve this :)

File: pkgs/development/python-modules/python-ldap/default.nix (hunk 3 above)
Comment (atalii) [reply pkgs/development/python-modules/python-ldap/default.nix:75]: Good to know. Sorry about that again!

**Review by atalii — APPROVED** [review]

File: pkgs/development/python-modules/python-ldap/default.nix (hunk 3 above)
Comment (mweinelt) [reply pkgs/development/python-modules/python-ldap/default.nix:75]: No worries!

**Comment by mweinelt** [comment]
> Should this be backported to 23.05?

We can, but we should rerun nixpkgs-review for 23.05 again.

**Comment by atalii** [comment]
Result of `nixpkgs-review pr 268179` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>odoo</li>
    <li>odoo.dist</li>
    <li>odoo15</li>
    <li>odoo15.dist</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>powerdns-admin</li>
  </ul>
</details>
<details>
  <summary>25 packages built:</summary>
  <ul>
    <li>adenum</li>
    <li>autofs5</li>
    <li>etebase-server (python311Packages.etebase-server)</li>
    <li>freeipa</li>
    <li>ldapmonitor</li>
    <li>python310Packages.django-auth-ldap</li>
    <li>python310Packages.django-auth-ldap.dist</li>
    <li>python310Packages.dogtag-pki</li>
    <li>python310Packages.dogtag-pki.dist</li>
    <li>python310Packages.etebase-server</li>
    <li>python310Packages.ldappool</li>
    <li>python310Packages.ldappool.dist</li>
    <li>python310Packages.python-ldap</li>
    <li>python310Packages.python-ldap.dist</li>
    <li>python311Packages.django-auth-ldap</li>
    <li>python311Packages.django-auth-ldap.dist</li>
    <li>python311Packages.dogtag-pki</li>
    <li>python311Packages.dogtag-pki.dist</li>
    <li>python311Packages.ldappool</li>
    <li>python311Packages.ldappool.dist</li>
    <li>python311Packages.python-ldap</li>
    <li>python311Packages.python-ldap.dist</li>
    <li>silenthound</li>
    <li>sssd</li>
    <li>tandoor-recipes</li>
  </ul>
</details>

**Comment by atalii** [comment]
Do you know anything about powerdns-admin? I'm not sure if that failure's important.

**Comment by mweinelt** [comment]
I know that the failure already exists on master, and is unrelated to this change.

**Pull request merged**